Create MQManager

nayuta28 opened the pull request:
Resolves #48.
## Purpose of this pull request

MQ接続の共通クラスを作成しました。
## Changes proposed in this pull request
- MessageQueueManagerクラス追加
- build.gradleに依存ライブラリ追加
## Issue Links

@javamas/javamas

nayuta28 committed:
Create MQManager

nayuta28 commented:
MQ共通クラス実装しましたのでプルリクさせて頂きました！
ローカルでサンプルのオブジェクトデータの送受信までは確認しています。
なんかもう色々ひどいんですけどグダグダするより上げて見てもらおうと思いました！
宜しくお願い致します。

dodoichi commented:
ざっと全体を眺めて、最低限修正して欲しいポイントを挙げておきます。
- フィールド変数の使用を控えてください。すべて不要です。
- すべての例外が握りつぶされていますが、これだと異常に気づくことができません。

あと、コードスタイルは事前に決めてなかったのですが、Googleのチェックをもとにスタイルをチェックしましょう。このブランチにプッシュしておきます。警告が出ても、すぐに直さなくてよいです。

dodoichi committed:
add basic checks.

CI checks code style. The checkstyle.xml is based on
[Google Checksi](https://google.github.io/styleguide/javaguide.html).

dodoichi committed:
use oracle jdk 8 to compile sources.

The setting of Travis CI has been modified.

dodoichi commented:
Travis CIの設定漏れで、一度ビルドに失敗しましたが、現在は問題ありません。
m(__)m

disc99 commented on build.gradle:
```diff
@@ -16,6 +17,13 @@ repositories {
 dependencies {
 
     testCompile 'junit:junit:4.+'
+    compile group: 'com.rabbitmq', name: 'amqp-client', version: '2.3.1'
```
せっかくなら、最新版を使っていいんじゃないかな？

disc99 commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+	 * @param queueName
+	 * @throws IOException
+	 */
+	public <T> T recieveMessage(Class<T> dto, String queueName) throws IOException {
```
Consumerのような、コールバックを引数で渡すイメージだったけど、この作りの場合は使う側はどのように呼び出すようになるんだろう？
簡単にでいいので、テストコードがあると使い方が分かりやすいかも！
あと、Travisでテストが通ることも確認できるｗ

HomMarkHunt commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+			ex.printStackTrace();
+		}
+	}
+}
```
Nit: 
- 最終行に改行を入れる

という意見が以前あった気が。

dodoichi committed:
add Sputnik integration.

Now the following static code analysis is configured.
- Checkstyle
- FindBugs
- PMD

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
```
[Checkstyle] WARNING: Import statement for 'com.fasterxml.jackson.core.JsonProcessingException' is in the wrong order. Should be in the 'THIRD_PARTY_PACKAGE' group, expecting not assigned imports on this line.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
```
[Checkstyle] WARNING: Import statement for 'com.fasterxml.jackson.databind.ObjectMapper' is in the wrong order. Should be in the 'THIRD_PARTY_PACKAGE' group, expecting not assigned imports on this line.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.rabbitmq.client.Channel;
```
[Checkstyle] WARNING: Import statement for 'com.rabbitmq.client.Channel' is in the wrong order. Should be in the 'THIRD_PARTY_PACKAGE' group, expecting not assigned imports on this line.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
```
[Checkstyle] WARNING: Import statement for 'com.rabbitmq.client.Connection' is in the wrong order. Should be in the 'THIRD_PARTY_PACKAGE' group, expecting not assigned imports on this line.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
```
[Checkstyle] WARNING: Import statement for 'com.rabbitmq.client.ConnectionFactory' is in the wrong order. Should be in the 'THIRD_PARTY_PACKAGE' group, expecting not assigned imports on this line.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import com.rabbitmq.client.QueueingConsumer;
```
[Checkstyle] WARNING: Import statement for 'com.rabbitmq.client.QueueingConsumer' is in the wrong order. Should be in the 'THIRD_PARTY_PACKAGE' group, expecting not assigned imports on this line.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import com.rabbitmq.client.QueueingConsumer;
+import com.rabbitmq.client.QueueingConsumer.Delivery;
```
[Checkstyle] WARNING: Import statement for 'com.rabbitmq.client.QueueingConsumer.Delivery' is in the wrong order. Should be in the 'THIRD_PARTY_PACKAGE' group, expecting not assigned imports on this line.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+ */
+public class MessageQueueManager {
+
+	private ConnectionFactory factory = null;
```
[Checkstyle] WARNING: 'member def modifier' have incorrect indentation level 8, expected level should be 2.

sputnikci replied:
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+public class MessageQueueManager {
+
+	private ConnectionFactory factory = null;
+	private Connection connection = null;
```
[Checkstyle] WARNING: 'member def modifier' have incorrect indentation level 8, expected level should be 2.

sputnikci replied:
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+
+	private ConnectionFactory factory = null;
+	private Connection connection = null;
+	private Channel channel = null;
```
[Checkstyle] WARNING: 'member def modifier' have incorrect indentation level 8, expected level should be 2.

sputnikci replied:
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+	private ConnectionFactory factory = null;
+	private Connection connection = null;
+	private Channel channel = null;
+	private ObjectMapper mapper = new ObjectMapper();
```
[Checkstyle] WARNING: 'member def modifier' have incorrect indentation level 8, expected level should be 2.

sputnikci replied:
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+	private Channel channel = null;
+	private ObjectMapper mapper = new ObjectMapper();
+
+	/**
```
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+	private ObjectMapper mapper = new ObjectMapper();
+
+	/**
+	 * RabbitMQにメッセージを送信する.
```
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+
+	/**
+	 * RabbitMQにメッセージを送信する.
+	 * メッセージのフォーマットはJSON.
```
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+	/**
+	 * RabbitMQにメッセージを送信する.
+	 * メッセージのフォーマットはJSON.
+	 * @param object
```
[Checkstyle] WARNING: At-clause should have a non-empty description.

sputnikci replied:
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+	 * RabbitMQにメッセージを送信する.
+	 * メッセージのフォーマットはJSON.
+	 * @param object
+	 * @param queueName
```
[Checkstyle] WARNING: At-clause should have a non-empty description.

sputnikci replied:
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+	 * メッセージのフォーマットはJSON.
+	 * @param object
+	 * @param queueName
+	 */
```
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+	 * @param object
+	 * @param queueName
+	 */
+	public void sendMessage(Object object, String queueName) {
```
[Checkstyle] WARNING: 'method def modifier' have incorrect indentation level 8, expected level should be 2.

sputnikci replied:
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+	 * @param queueName
+	 */
+	public void sendMessage(Object object, String queueName) {
+		try {
```
[Checkstyle] WARNING: 'try' have incorrect indentation level 16, expected level should be 4.

sputnikci replied:
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+	 */
+	public void sendMessage(Object object, String queueName) {
+		try {
+			connectMessageQueue();
```
[Checkstyle] WARNING: 'try' child have incorrect indentation level 24, expected level should be 6.

sputnikci replied:
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+	public void sendMessage(Object object, String queueName) {
+		try {
+			connectMessageQueue();
+			setQueue(queueName);
```
[Checkstyle] WARNING: 'try' child have incorrect indentation level 24, expected level should be 6.

sputnikci replied:
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+		try {
+			connectMessageQueue();
+			setQueue(queueName);
+			String json = mapper.writeValueAsString(object);
```
[Checkstyle] WARNING: 'member def type' have incorrect indentation level 24, expected level should be 6.

sputnikci replied:
[Checkstyle] WARNING: 'try' child have incorrect indentation level 24, expected level should be 6.

sputnikci replied:
[Checkstyle] WARNING: Line contains a tab character.

sputnikci commented on src/main/java/com/javamas/araignee/mq/MessageQueueManager.java:
```diff
@@ -0,0 +1,96 @@
+			connectMessageQueue();
+			setQueue(queueName);
+			String json = mapper.writeValueAsString(object);
+			channel.basicPublish("", queueName, null, json.getBytes());
```
[Checkstyle] WARNING: 'try' child have incorrect indentation level 24, expected level should be 6.

sputnikci replied:
[Checkstyle] WARNING: Line contains a tab character.
